<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <arg name="sub_lidar_topic_name" default="/velodyne_packets"/>
  <arg name="sub_twist_topic_name" default="/can_twist"/>
  <arg name="sub_imu_topic_name" default="/imu/data_raw"/>
  <arg name="reverse_imu" default="false"/>
  <arg name="velocity_scale_factor" default="1.0"/>
  <arg name="slip_sf" default="1.0"/>
  <arg name="slip_probability" default="1"/>
  <arg name="missing_can_probability" default="1"/>

  <remap from="sub_gnss_topic_name" to="/navsat/fix" />

  <arg name="pub_lidar_topic_name" default="/sensing/lidar/top/velodyne_packets"/>
  <arg name="pub_twist_topic_name" default="/vehicle/status/twist"/>
  <arg name="pub_imu_topic_name" default="/sensing/imu/imu_data"/>

  <param name="sub_lidar_topic_name" value="$(arg sub_lidar_topic_name)"/>
  <param name="sub_twist_topic_name" value="$(arg sub_twist_topic_name)"/>
  <param name="sub_imu_topic_name" value="$(arg sub_imu_topic_name)"/>
  <param name="pub_lidar_topic_name" value="$(arg pub_lidar_topic_name)"/>
  <param name="pub_twist_topic_name" value="$(arg pub_twist_topic_name)"/>
  <param name="pub_imu_topic_name" value="$(arg pub_imu_topic_name)"/>
  <param name="reverse_imu" value="$(arg reverse_imu)"/>
  <param name="velocity_scale_factor" value="$(arg velocity_scale_factor)"/>
  <param name="slip_sf" value="$(arg slip_sf)"/>
  <param name="slip_probability" value="$(arg slip_probability)"/>
  <param name="missing_can_probability" value="$(arg missing_can_probability)"/>

  <node pkg="type_changer_real" name="type_changer_real" type="type_changer_real"/>
  <node pkg="topic_tools" type="relay" name="gnss_relay" args="sub_gnss_topic_name /sensing/gnss/nav_sat_fix" />

</launch>
